src/PeerPoolMgr.h

```diff
@@ -18,6 +18,30 @@ class HttpRequest;
 class CachePeer;
 class CommConnectCbParams;
 
+/// CodeContext for cache_peer related classes
+class DetailedCodeContext : public CodeContext
```

Comment:
Assuming other change requests in this review are correct:

```suggestion
/// CodeContext with constant details known at construction time
class PrecomputedCodeContext: public CodeContext
```

And move this declaration to src/base/PrecomputedCodeContext.{h,cc}.

Reply:
Done.